zerver/views/streams.py

```diff
@@ -648,6 +650,164 @@ def you_were_just_subscribed_message(
 RETENTION_DEFAULT: str | int = "realm_default"
 
 
+@transaction.atomic(savepoint=False)
+@require_non_guest_user
+@typed_endpoint
+def add_channel(
```

Comment:
I think `create_channel` would be a better name, to align with all the other code for this endpoint.